cggmp21/Cargo.toml

```diff
@@ -47,10 +47,12 @@ generic-tests = { workspace = true }
 
 [features]
 all-curves = ["curve-secp256k1", "curve-secp256r1", "curve-stark"]
-curve-secp256k1 = ["generic-ec/curve-secp256k1"]
-curve-secp256r1 = ["generic-ec/curve-secp256r1"]
-curve-stark = ["generic-ec/curve-stark"]
+curve-secp256k1 = ["generic-ec/curve-secp256k1", "hd-wallet?/curve-secp256k1"]
+curve-secp256r1 = ["generic-ec/curve-secp256r1", "hd-wallet?/curve-secp256r1"]
+curve-stark = ["generic-ec/curve-stark", "hd-wallet?/curve-stark"]
 hd-wallet = ["dep:hd-wallet", "cggmp21-keygen/hd-wallet"]
+hd-slip10 = ["hd-wallet/slip10"]
+hd-stark = ["hd-wallet/stark"]
```

Comment:
What do you think about enabling the hd wallet features as soon as hd-wallet + corresponding curve is enabled? That is, making ` curve-secp256k1 = ["generic-ec/curve-secp256k1", "hd-wallet?/curve-secp256k1", "hd-wallet?/slip10"]` and `curve-stark = ["generic-ec/curve-stark", "hd-wallet?/curve-stark", "hd-wallet?/stark"]`

Reply:
Where possible, it's already implied by `hd-wallet` crate itself, see in [Cargo.toml](https://github.com/LFDT-Lockness/hd-wallet/blob/m/Cargo.toml#L31-L38): 

```toml
# Adds support of secp256k1 curve to slip10 derivation
curve-secp256k1 = ["generic-ec/curve-secp256k1", "slip10"]
# Adds support of secp256r1 curve to slip10 derivation
curve-secp256r1 = ["generic-ec/curve-secp256r1", "slip10"]
# Enables Edwards-specific derivation
curve-ed25519 = ["generic-ec/curve-ed25519", "edwards"]
# Enables Stark-specific derivation
curve-stark = ["generic-ec/curve-stark", "stark"]
```